### ISSUE TITLE
Some questions about RSSS.py

### ISSUE THREAD

**Issue by WesternTrail** [Cwuwhu/FCD-GAN-pytorch#3]
When I run the code, the following error occurs:

![image](https://user-images.githubusercontent.com/28615285/223601684-a27d2a08-8be6-428a-9db2-9ab9efc49a4d.png)

![image](https://user-images.githubusercontent.com/28615285/223602105-ba99d4c0-91a7-44af-b2ad-01fad74ebd01.png)

I suspect that the download [oscd dataset] (https://ieee-dataport.org/open-access/oscd-onera-satellite-change-detection) has some problems. Can the author share the download address of OSCD dataset??

**Comment by autumoon** [comment]
Hello, May I ask for the requirements.txt of your envs just like below? I am just starting the projects while i don't know the requirements excatly, 3q.

`matplotlib~=3.6.2
numpy~=1.23.4
pillow~=9.3.0
pyyaml~=6.0
scikit-learn~=1.1.3
torchinfo~=1.7.1
geopandas~=0.12.1
pandas~=1.5.1
timm~=0.6.12
yacs~=0.1.8
seaborn~=0.12.0
lightgbm~=3.3.4
tqdm~=4.64.1
PyMySQL~=1.0.2
minio~=7.1.13
opencv-python~=4.7.0.68
scikit-image~=0.19.3
einops~=0.6.0
requests~=2.28.1
Flask~=2.2.2
onnx~=1.13.0`

**Comment by WesternTrail** [comment]
This is my environment, as well as the environment for changeformer and BIT. There may be redundancy

`absl-py==1.4.0
albumentations==1.3.0
cachetools==5.3.0
certifi @ file:///C:/b/abs_85o_6fm0se/croot/certifi_1671487778835/work/certifi
charset-normalizer==3.0.1
colorama==0.4.6
contourpy==1.0.7
cycler==0.11.0
einops==0.3.2
ever-beta==0.2.3
filelock==3.9.0
fonttools==4.38.0
GDAL @ file:///C:/Users/%E9%99%88%E9%93%AD/Downloads/GDAL-3.4.3-cp39-cp39-win_amd64.whl
google-auth==2.16.1
google-auth-oauthlib==0.4.6
grpcio==1.51.3
huggingface-hub==0.12.1
idna==3.4
imageio==2.26.0
importlib-metadata==6.0.0
importlib-resources==5.12.0
joblib==1.2.0
kiwisolver==1.4.4
lazy_loader==0.1
Markdown==3.4.1
MarkupSafe==2.1.2
matplotlib==3.7.0
networkx==3.0
numpy==1.24.2
oauthlib==3.2.2
opencv-python-headless==4.7.0.72
packaging==23.0
pandas==1.5.3
Pillow==9.4.0
prettytable==3.6.0
protobuf==4.22.0
pyasn1==0.4.8
pyasn1-modules==0.2.8
pyparsing==3.0.9
python-dateutil==2.8.2
pytz==2022.7.1
PyWavelets==1.4.1
PyYAML==6.0
qudida==0.0.4
requests==2.28.2
requests-oauthlib==1.3.1
rsa==4.9
scikit-image==0.20.0
scikit-learn==1.2.1
scipy==1.9.1
six==1.16.0
tensorboard==2.12.0
tensorboard-data-server==0.7.0
tensorboard-plugin-wit==1.8.1
threadpoolctl==3.1.0
tifffile==2023.2.28
timm==0.6.12
torch==1.12.1+cu113
torchaudio==0.12.1+cu113
torchvision==0.13.1+cu113
tqdm==4.64.1
typing_extensions==4.5.0
urllib3==1.26.14
wcwidth==0.2.6
Werkzeug==2.2.3
wincertstore==0.2
zipp==3.15.0
`

**Comment by autumoon** [comment]
> This is my environment, as well as the environment for changeformer and BIT. There may be redundancy
> 
> `absl-py==1.4.0 albumentations==1.3.0 cachetools==5.3.0 certifi @ file:///C:/b/abs_85o_6fm0se/croot/certifi_1671487778835/work/certifi charset-normalizer==3.0.1 colorama==0.4.6 contourpy==1.0.7 cycler==0.11.0 einops==0.3.2 ever-beta==0.2.3 filelock==3.9.0 fonttools==4.38.0 GDAL @ file:///C:/Users/%E9%99%88%E9%93%AD/Downloads/GDAL-3.4.3-cp39-cp39-win_amd64.whl google-auth==2.16.1 google-auth-oauthlib==0.4.6 grpcio==1.51.3 huggingface-hub==0.12.1 idna==3.4 imageio==2.26.0 importlib-metadata==6.0.0 importlib-resources==5.12.0 joblib==1.2.0 kiwisolver==1.4.4 lazy_loader==0.1 Markdown==3.4.1 MarkupSafe==2.1.2 matplotlib==3.7.0 networkx==3.0 numpy==1.24.2 oauthlib==3.2.2 opencv-python-headless==4.7.0.72 packaging==23.0 pandas==1.5.3 Pillow==9.4.0 prettytable==3.6.0 protobuf==4.22.0 pyasn1==0.4.8 pyasn1-modules==0.2.8 pyparsing==3.0.9 python-dateutil==2.8.2 pytz==2022.7.1 PyWavelets==1.4.1 PyYAML==6.0 qudida==0.0.4 requests==2.28.2 requests-oauthlib==1.3.1 rsa==4.9 scikit-image==0.20.0 scikit-learn==1.2.1 scipy==1.9.1 six==1.16.0 tensorboard==2.12.0 tensorboard-data-server==0.7.0 tensorboard-plugin-wit==1.8.1 threadpoolctl==3.1.0 tifffile==2023.2.28 timm==0.6.12 torch==1.12.1+cu113 torchaudio==0.12.1+cu113 torchvision==0.13.1+cu113 tqdm==4.64.1 typing_extensions==4.5.0 urllib3==1.26.14 wcwidth==0.2.6 Werkzeug==2.2.3 wincertstore==0.2 zipp==3.15.0 `

3q so much，老铁，best wishes

**Comment by Cwuwhu** [comment]
Thank your following our work. The OSCD dataset used in the code was preprocessed by us. This dataset of OSCD comprised 24 pairs of multi-temporal images, acquired by Sentinel-2 satellite. It contains 14 training and 10 test image pairs. The remote sensing imagery from Sentinel-2 satellite has 13 spectral bands with the spatial resolution of 10, 20m and 60. For flexi-ble process, we only choose four spectral bands (B, G, R and NIR) with 10m resolution. Co-registration was im-plemented to reduce the mis-registration errors in some pairs, and common subsets were selected to keep the same size of multi-temporal image pair and reference.

**Comment by WesternTrail** [comment]
@Cwuwhu thank you for your answer!

**Comment by WesternTrail** [comment]
Can provide a pre processing program? So that we can run the RSSS.py file normally.thank you!

**Comment by 15738281321** [comment]
> Thank your following our work. The OSCD dataset used in the code was preprocessed by us. This dataset of OSCD comprised 24 pairs of multi-temporal images, acquired by Sentinel-2 satellite. It contains 14 training and 10 test image pairs. The remote sensing imagery from Sentinel-2 satellite has 13 spectral bands with the spatial resolution of 10, 20m and 60. For flexi-ble process, we only choose four spectral bands (B, G, R and NIR) with 10m resolution. Co-registration was im-plemented to reduce the mis-registration errors in some pairs, and common subsets were selected to keep the same size of multi-temporal image pair and reference.

可以分享一下数据吗？
Can you share the data?

**Comment by Cwuwhu** [comment]
> Can provide a pre processing program? So that we can run the RSSS.py file normally.thank you!

You can use 'OSCDProcess.py'
